src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,86 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+import React from 'react';
+
+interface Props {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void,
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const numberPag = Math.ceil(total / perPage);
+
+  function countPagination() {
+    const countArr: number[] = [];
+
+    for (let i = 1; i <= numberPag; i += 1) {
+      countArr.push(i);
+    }
+
+    return countArr;
+  }
+
+  const count = countPagination();
+
+  function changeCheck(numPage: number) {
```

Comment:
The 'changeCheck' function can be replaced with a useCallback hook to avoid unnecessary re-rendering.